symforce/pybind/lcm_type_casters.h

```diff
@@ -72,7 +72,7 @@ struct lcm_type_caster {
         py::module_::import(module_path.c_str()).attr(LCMType::getTypeName());
     py::object result = lcm_py_type.attr("decode")(msg_bytes);
     result.inc_ref();
-    return result;
+    return std::move(result);
```

Comment:
This here is kind of strange.
I got the error (well, warning, but warnings are errors now)
```
      In file included from /Users/runner/work/symforce/symforce/symforce/pybind/cc_key.cc:13:
      /Users/runner/work/symforce/symforce/symforce/pybind/./lcm_type_casters.h:75:12: error: local variable 'result' will be copied despite being returned by name [-Werror,-Wreturn-std-move]
          return result;
                 ^~~~~~
      /Users/runner/work/symforce/symforce/symforce/pybind/./lcm_type_casters.h:75:12: note: call 'std::move' explicitly to avoid copying
          return result;
                 ^~~~~~
                 std::move(result)
```
It only appears on the macos runner (I couldn't reproduce it --- though I didn't try super hard) on an M1 I have on hand.

The compiler being used was
```
The CXX compiler identification is AppleClang 13.0.0.13000029
```
Here's a link to an example job: https://github.com/bradley-solliday-skydio/symforce/actions/runs/3109686309/jobs/5040132490

Note sure why the compiler thinks a copy can't be elided. Note sure if the other compilers are silently not eliding the return value.

But, never minding the reason for now, I figured moving an object as lightweight as a `py::object` should be harmless, and a user might be using this compiler (if that's the cause of the problems), so might as well make it happy.

It does raise the question though: A user might be sad if they can't build symforce (easily) because a compiler warning which appears only on their build setup was made into an error.

I'm not sure what the default behavior is, but it might be good to not have warnings as errors by default, but to instead simply turn it on for CI and the like. That way we get the benefits of fixing the warnings as soon as they pop up while developing, but without possibly causing trouble for a user.

Reply:
Is your M1 also running Clang 13?

> Note sure why the compiler thinks a copy can't be elided.

Not sure why it _can't_, but the standard does not guarantee NRVO even in C++17 (and doesn't guarantee any copy elision until C++17), so it's certainly not required to. 

The default is to treat warnings as errors, see `CMakeLists.txt:73` - you can also set `SYMFORCE_COMPILE_OPTIONS=""` if you don't want this.  I think I agree it makes sense to not set `-Werror` by default, but not sure if there's a clean idiomatic way to make sure it's on everywhere in CI

Reply:
The M1 is running `AppleClang 13.1.6.13160021`